content/techniques/caching.md

```diff
@@ -161,6 +161,63 @@ import { AppController } from './app.controller';
 export class ApplicationModule {}
 ```
 
+##### Redis cache `set` and `get` example
+
+First install the type dependency:
+
+```bash
+$ npm install --save-dev @types/cache-manager
+```
+
+```typescript
+// AppService.ts
+import { Injectable, Inject, CACHE_MANAGER } from '@nestjs/common';
+import { Cache } from 'cache-manager';
+
+@Injectable()
+export class AppService {
+  constructor(@Inject(CACHE_MANAGER) private cacheManager: Cache) {
+  }
+
+  async getHello(): Promise<string> {
+    // callback way
+    this.cacheManager.set('foo', 'bar', {ttl: 100000}, (err) => {
+      this.cacheManager.get('foo', (errGet, result) => {
+        console.log(result); // logs "bar"
+      });
+    });
+
+    // async/await way
+    await this.cacheManager.set('foo2', 'bar2', {ttl: 100000});
+    const res =  await this.cacheManager.get('foo2');
+    console.log(res); // logs "bar2"
+
+    return 'Hello World!';
+  }
+}
+```
+
+Storage in Redis cache can be also be verified by [redis-cli](https://redis.io/topics/rediscli) or Redis's desktop client such as [Redis desktop manager](https://redisdesktop.com/).
+
+As this method `getHello()` of `AppService.ts` file is returning `Promise<string>`, return type of controller using this service should also be changed:
+
+```typescript
+import { Controller, Get } from '@nestjs/common';
+import { AppService } from './app.service';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @Get()
+  getHello(): Promise<string> {
+    return this.appService.getHello();
+  }
+}
+```
+
+Visit [docs](https://github.com/dabroek/node-cache-manager-redis-store) of Redis store for full list API usage.
+
```

Comment:
Is this even relevant here? I think we can remove it

Reply:
Yes, not very relavant here. But I think beginners/new comers will face issues. Generally controller return string or array and if async/await way is not used in service. When async/await is used in service, the return type of controller should be also changed, so I added that docs

Reply:
This isn't relevant to Caching though. Such an issue could appear by following every chapter in the documentation. Can you please remove this part? We can't even assume that the developer has `AppController` at this point.

Reply:
removed in 150e83d7d8263e3dbd9857c948f6cd1306eef81f

Reply:
Thanks!